src/query/dblink_scan.c

```diff
@@ -301,6 +333,147 @@ dblink_make_date_time_tz (T_CCI_U_TYPE utype, DB_VALUE * value_p, T_CCI_DATE_TZ
+	    }
+	  break;
+	case DB_TYPE_STRING:
+	case DB_TYPE_VARNCHAR:
+	case DB_TYPE_CHAR:
+	case DB_TYPE_NCHAR:
+	  a_type = CCI_A_TYPE_STR;
+	  u_type = CCI_U_TYPE_STRING;
+	  value = (void *) vd->dbval_ptr[i].data.ch.medium.buf;
+	  break;
+	case DB_TYPE_DATE:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_DATE;
+	  db_date_decode ((DB_DATE *) value, &month, &day, &year);
+	  cci_date.mon = month;
+	  cci_date.day = day;
+	  cci_date.yr = year;
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_TIME:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_TIME;
+	  db_time_decode (&vd->dbval_ptr[i].data.time, &hh, &mm, &ss);
+	  cci_date.hh = hh;
+	  cci_date.mm = mm;
+	  cci_date.ss = ss;
+	  cci_date.ms = 0;
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_DATETIME:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_DATETIME;
+	  DATETIME_DECODE (cci_date, vd->dbval_ptr[i].data.datetime, month, day, year, hh, mm, ss, ms);
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_TIMESTAMP:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_TIMESTAMP;
+	  timestamp = &vd->dbval_ptr[i].data.utime;
+	  db_timestamp_decode_ses (timestamp, &date, &time);
+	  TIMESTAMP_DECODE (cci_date, date, time, month, day, year, hh, mm, ss);
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_TIMESTAMPTZ:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_TIMESTAMPTZ;
+	  timestamp = &vd->dbval_ptr[i].data.timestamptz.timestamp;
+	  zone_id = &vd->dbval_ptr[i].data.timestamptz.tz_id;
+	  db_timestamp_decode_w_tz_id (timestamp, zone_id, &date, &time);
+	  TIMESTAMP_DECODE (cci_date, date, time, month, day, year, hh, mm, ss);
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_TIMESTAMPLTZ:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_TIMESTAMPLTZ;
+	  timestamp = &vd->dbval_ptr[i].data.timestamptz.timestamp;
+	  db_timestamp_decode_utc (timestamp, &date, &time);
+	  TIMESTAMP_DECODE (cci_date, date, time, month, day, year, hh, mm, ss);
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_DATETIMETZ:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_DATETIMETZ;
+	  datetime = &vd->dbval_ptr[i].data.datetimetz.datetime;
+	  zone_id = &vd->dbval_ptr[i].data.datetimetz.tz_id;
+	  tz_utc_datetimetz_to_local (datetime, zone_id, &dt_local);
+	  DATETIME_DECODE (cci_date, dt_local, month, day, year, hh, mm, ss, ms);
+	  value = &cci_date;
+	  break;
+	case DB_TYPE_DATETIMELTZ:
+	  a_type = CCI_A_TYPE_DATE;
+	  u_type = CCI_U_TYPE_DATETIMELTZ;
+	  datetime = &vd->dbval_ptr[i].data.datetimetz.datetime;
+	  tz_datetimeltz_to_local (datetime, &dt_local);
+	  DATETIME_DECODE (cci_date, dt_local, month, day, year, hh, mm, ss, ms);
+	  value = &cci_date;
+	  break;
+	default:
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_DBLINK, 1, "bind: not supported type");
+	  return S_ERROR;
+	}
+      ret = cci_bind_param (scan_info->stmt_handle, i + 1, a_type, value, u_type, 0);
+      if (ret < 0)
+	{
+	  if (ret == CCI_ER_BIND_INDEX)
+	    {
+	      break;
```

Comment:
S_SUCCESS is returned when exiting from here with a break.
Is your intention correct?

Reply:
yes, return S_SUCCESS at the outside of this switch.